### PR TITLE
More draw handling.

### PR DESCRIPTION
**Pull request by TFiFiE**
If the use of different komi is going to be supported, then that should include integral ones, and that means the possibility of draws. I restrained myself however from introducing a contempt factor. :smile:

### CODE REVIEW

File: src/UCTSearch.cpp [hunk 1]
```diff
@@ -413,7 +413,7 @@ int UCTSearch::get_best_move(passflag_t passflag) {
     assert(first_child != nullptr);
 
     auto bestmove = first_child->get_move();
-    auto bestscore = first_child->get_eval(color);
+    auto bestscore = first_child->get_pure_eval(color);
```
Comment (gcp) [inline src/UCTSearch.cpp]: No guarantee yet the first child has visits?

Comment (TFiFiE) [reply src/UCTSearch.cpp]: Does that guarantee exist with `get_eval` though? Maybe make it `0.5f` or `1.0f` with no visits?

Comment (gcp) [reply src/UCTSearch.cpp]: Your point is right, but get_eval works by coincidence due to virtual loss and so won't ever crash here, which the new code would.

Comment (TFiFiE) [reply src/UCTSearch.cpp]: I've changed it, but I don't see how `first_child` is guaranteed to have a virtual loss.

Comment (gcp) [reply src/UCTSearch.cpp]: Hmm, it's not (it's actually guaranteed not to in get_best_move). But if it isn't, then changing the calls from get_eval to get_pure_eval is a no-op obviously.

Maybe it's good to check what exactly happens with p = 1 and see if we get the edge cases right? And is specifying p = 0 possible?

**Comment by alreadydone** [comment]
With maximizing winrate and not the score as the goal, drawn games may be the only chances for us to see AIs play perfect endgames ...

**Pull request merged**